CONTRIBUTING.md

```diff
@@ -3,20 +3,46 @@
-1. Fork the repo
-2. git clone your fork
-3. Create feature branch
-4. Make and checkin your changes along with unit tests
-5. git commit your changes
-6. git push origin HEAD
-7. To request merge into master send a pull request from the web ui
-https://github.com/Microsoft/onnxruntime.
-8. Add 'Microsoft/onnxruntime' as a reviewer.
-
-New code *must* be accompanied by unit tests.
-
-*Note*: After creating a pull request, you might not see a build getting triggered right away. One of the
+# Proposing new public APIs
+The ONNX runtime has a collection of [public APIs](docs/HighLevelDesign.md).  Some of these API's make their way back into the Windows OS.  We make compatibility committments for these API's and follow a structured process when adding to them.  Please use the [Feature Request issue template](issues/new?template=feature_request.md) before starting any PR's that affect any of the public API's.
+
+# Process details
+
+Please search the [issue tracker](https://github.com/microsoft/onnxruntime/issues) for a similar idea first: there may already be an issue you can contribute to.
+
+1. **Create Issue**  
+To propose a new feature or API please start by filing a new issue in the [issue tracker](https://github.com/microsoft/onnxruntime/issues).  
+Include as much detail as you have. It's fine if it's not a complete design: just a summary and rationale is a good starting point.
+
+2. **Discussion**  
+We'll keep the issue open for community discussion until it's been resolved or is deemed no longer relevant.
+Note that if an issue isn't a high priority or has many open questions then it might stay open for a long time.
+
+3. **Owner Review**  
+The ONNX Runtime team will review the proposal and either approve or close the issue based on whether it broadly aligns with the [Onnx Runtime Roadmap - High Level Goals section](../docs/Roadmap.md) and contribution guidelines.
+
+4. **API Review**  
+If the feature adds new APIs then we'll start an API review.
+All new public APIs must be reviewed before merging.  
+For making changes to the C API, refer to guidance [here](onnxruntime/core/session/onnxruntime_c_api.cc#L1326).
+For making changes to the WinRT API, please work directly with the owner assigned in step 3.
+
+5. **Implementation**
+* A feature can be implemented by you, the ONNX Runtime team, or other community members.  Code contributions are greatly appreciated: feel free to work on any reviewed feature you proposed, or choose one in the backlog and send us a PR. If you are new to the project and want to work on an existing issue, we recommend starting with issues that are tagged with “good first issue”. Please let us know in the issue comments if you are actively working on implementing a feature so we can ensure it's assigned to you.  
+* New code *must* be accompanied by unit tests.
+* Documentation and sample updates: The feature owner updates the documentation and if applicable add a sample to the samples repository.
```

Comment:
can you call out specifically that "the feature owner is responsible for making updates"  so there is no ambiguity on whether someone from ORT team will do this vs the contributor?